### PR TITLE
BoundPyOpenCLProgram: do not store a queue

### PR DESCRIPTION
**Pull request by inducer**
@kaushikcfd I suspect this'll cause some breakage on your end of things (sorry!). But I think it's probably better to get this out of the way sooner rather than later.

### CODE REVIEW

File: pytato/target/loopy/codegen.py [hunk 1]
```diff
@@ -652,7 +656,8 @@ def get_initial_codegen_state(namespace: Namespace, target: LoopyTarget,
 
 def generate_loopy(result: Union[Array, DictOfNamedArrays, Dict[str, Array]],
         target: Optional[LoopyTarget] = None,
-        options: Optional[lp.Options] = None) -> BoundProgram:
+        options: Optional[lp.Options] = None,
+        cl_device: Optional["pyopencl.Device"] = None) -> BoundProgram:
```
Comment (kaushikcfd) [inline pytato/target/loopy/codegen.py:660]: cl_device should be passed via target. Passing cl_device to generate_loopy makes generate_loopy CL-specific which it shouldn't be.

**Review by kaushikcfd — CHANGES_REQUESTED** [review]
I have one concern on which many other proposed changes depend.

File: pytato/target/loopy/codegen.py (hunk 1 above)
Comment (inducer) [reply pytato/target/loopy/codegen.py:661]: I see what you're saying, but I don't quite see it that way. The new interface isn't any less general than the one before this change: You can still pass `target` and use any target you like. Passing `cl_device` is just a shorthand for "use the CL target with this device", to win a few characters in a very frequently used function. (dabbea5 makes it explicit that you should either pass one or the other, but never both, and c52e205 makes `cl_device` kw-only).

Comment (inducer) [reply pytato/target/loopy/codegen.py:661]: Do you find that acceptable, or would you prefer the explicit target still?

**Review by kaushikcfd — APPROVED** [review]
LGTM. One minor concern, that's also optional for the merge. Thanks!

File: pytato/target/__init__.py [hunk 2]
```diff
@@ -73,14 +73,9 @@ class BoundProgram:
...
 
         A map from names to pre-bound kernel arguments.
-
-    .. automethod:: __call__
     """
     program: Any
     bound_arguments: Mapping[str, Any]
     target: Target
 
-    def __call__(self, *args: Any, **kwargs: Any) -> Any:
```
[the remark this answers is not in the document]
Comment (inducer) [reply pytato/target/__init__.py:83]: It is, but I feel that the interface will vary based on the subclass. (PyOpenCL needing a `queue` for example.) And mypy was unhappy about that, so the easiest way around that as to remove the interface from the base class. Got a better idea?

Bonus formatting tip:
```
`__call__` vs __call__
```
`__call__` vs __call__

[the remark this answers is not in the document]
Comment (kaushikcfd) [reply pytato/target/__init__.py:83]: Yep, reasonable. Can't seem to find anything better.

[the remark this answers is not in the document]
Comment (inducer) [reply pytato/target/__init__.py:83]: 35b6a4c adds docs to clarify.

**Comment by inducer** [comment]
@kaushikcfd Thanks for taking a look!

**Pull request merged**